jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh

```diff
@@ -0,0 +1,105 @@
+    kieResponse=$(query_ocp_api "apis/apps.openshift.io" "${kieServeruri}")
+    kieReplicas=$(echo ${kieResponse:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["items"][0]["spec"]["replicas"])')
+
+    controllerResponse=$(query_ocp_api "apis/apps.openshift.io" "deploymentconfigs/${WORKBENCH_SERVICE_NAME}")
+    controllerReplicas=$(echo ${controllerResponse:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["spec"]["replicas"])')
+    controllerAuth="$(echo -n "${KIE_ADMIN_USER}:${KIE_ADMIN_PWD}" | base64)"
+
+    if [ ${kieReplicas} == 0 ]; then
+        log_info "KIE Server Replicas is ${kieReplicas}, updating ${KIE_SERVER_ID} configMap to DETACHED."
+        update_config_map "{\"metadata\":{\"labels\":{\"services.server.kie.org/kie-server-state\":\"DETACHED\"}}}"
+
+        if [ "${controllerServiceHost}x" != "x" -a "${controllerServicePort}x" != "x" -a ${controllerReplicas} -gt 0  ]; then
+            # curl command may be hanging a bit during dc pod starting up; need to update
+            # Pod.spec.terminationGracePeriodSeconds so as to safe guard infinite wait
+            # and also accommodate bc Pod startup time.
+            # try 12 times waiting 10 seconds for new retry, max allowed time is 120s
+            curl --retry 12 --retry-delay 10 --retry-max-time 120 \
+                -s -X DELETE -i "http://${controllerServiceHost}:${controllerServicePort}/rest/controller/server/${KIE_SERVER_ID}" \
+                -H "Content-Type: application/json" \
+                -H "Authorization: Basic ${controllerAuth}"
+            log_info "Controller successfully notified"
+        fi
+
+    elif [ ${kieReplicas} -gt 0 ]; then
+        log_info "KIE Server Replicas is ${kieReplicas}, updating ${KIE_SERVER_ID} configMap to USED."
+        update_config_map "{\"metadata\":{\"labels\":{\"services.server.kie.org/kie-server-state\":\"USED\"}}}"
+
+        if [ "${controllerServiceHost}x" != "x" -a "${controllerServicePort}x" != "x" -a ${controllerReplicas} -gt 0  ]; then
+            # curl command may be hanging a bit during dc pod starting up; need to update
+            # Pod.spec.terminationGracePeriodSeconds so as to safe guard infinite wait
+            # and also accommodate bc Pod startup time.
+            # try 12 times waiting 10 seconds for new retry, max allowed time is 120s
+            curl --retry 12 --retry-delay 10 --retry-max-time 120 \
+                -s -X PUT -i "http://${controllerServiceHost}:${controllerServicePort}/rest/controller/server/${KIE_SERVER_ID}" \
+                -H "Content-Type: application/json" \
+                -H "Authorization: Basic ${controllerAuth}"
+            log_info "Controller successfully notified"
+        fi
+    fi
+else
+    log_warning "No Controller found or KIE_SERVER_ID is not set, skipping..."
+fi
+
+#if [ "${serviceHost}x" != "x" -a "${servicePort}x" != "x" ]; then
```

Comment:
@spolti Don't we want to remove all these commented lines?

Reply:
ops, removing it.